Xamarin.Essentials/Connectivity/Connectivity.android.cs

```diff
@@ -21,6 +22,38 @@ static void StartListeners()
             Platform.AppContext.RegisterReceiver(conectivityReceiver, new IntentFilter(ConnectivityManager.ConnectivityAction));
         }
 
+        static SignalStrength PlatformSignalStrength()
+        {
+            Permissions.EnsureDeclared(PermissionType.WifiState);
+            try
+            {
+                var wifiManager = Platform.WifiManager;
+                if (wifiManager == null)
+                    return SignalStrength.Unknown;
+
+                var info = wifiManager.ConnectionInfo;
+                if (info == null)
+                    return SignalStrength.None;
+
+                switch (WifiManager.CalculateSignalLevel(info.Rssi, 3))
```

Comment:
Looking at the docs, this function appears to return 0 - num-1, so in this case 0 - 2 and will never be strong.

I am thinking that we should do 5 levels: 0 = none, 1 = weak, 2 = fair, 3-4 = strong
I am going for 3-4 as strong as I don't think it is ever the case where the signal strength is at max, so will technically be 0.99 or something. Need to test that theory though.

Reply:
Yeah I am also not sure about when this will be max. I'll test this first, but for the time being I used your partitioning.

Reply:
Finished testing, currently all available values are reachable if you have the equivalent signal strength on android